nova/db/sqlalchemy/api.py

```diff
@@ -411,9 +411,26 @@ def convert_objects_related_datetimes(values, *datetime_keys):
 
 
 def _sync_instances(context, project_id, user_id):
-    return dict(zip(('instances', 'cores', 'ram'),
-                    _instance_data_get_for_user(context, project_id, user_id)))
-
+    query = '''
+      SELECT instance_type_id, COUNT(id), SUM(vcpus), SUM(memory_mb),
+        EXISTS(SELECT instance_type_id
+               FROM instance_type_extra_specs
+               WHERE instance_type_id = instances.instance_type_id
```

Comment:
That `instance_type_id` is unfortunately the primary key of `InstanceType`, which is used only internally and is a separate id then the `flavorid` field of `InstanceType` (which is public). This way, an instance can point to the "same" flavour, even when we update the flavour.

I would suggest to do a join, and try to tabularise the sum over the `quota_key_instances` as you do here:
https://github.com/sapcc/nova/pull/16/files#diff-77f9348ab09642ba46409b6828af4af0R934

Reply:
I think, that would probably avoid a couple of roundtrips to the db.

Reply:
Oh yeah, thanks for catching that. I forgot about this file when amending all the other usages of `instance_type_id` to use the `flavorid`.